Bio/SeqFeature.py

```diff
@@ -1534,10 +1527,7 @@ def extract(self, parent_sequence):
         """
         # This copes with mixed strand features & all on reverse:
         parts = [loc.extract(parent_sequence) for loc in self.parts]
-        # We use addition rather than a join to avoid alphabet issues:
-        f_seq = parts[0]
-        for part in parts[1:]:
-            f_seq += part
```

Comment:
This returns a string, not a Seq. How about:

```python
return Seq("").join(parts)
```

Reply:
Right. But the function should return an object of the same type as the input, which could be string, Seq, SeqRecord etc. I changed this to `functools.reduce`.